### DISCUSSION TITLE
Option to connect to hostnames instead of IP

### DISCUSSION THREAD

**Discussion by codenulls** [wiresock/proxifyre#73] [Q&A] [answered]
ProxiFyre will connect to the IP address directly instead of connecting to the hostname. When I'm using a proxy service, they complain about not being able to see the hostnames. 

Let's say, if I visit any website in firefox, it doesn't connect to the hostname, but rather the IP. 

This is not a high priority issue, but a feature like this would be very nice. Thank you for providing such a great tool for free!

**Comment by wiresock** [comment, marked as answer]
Thank you for the kind words and feedback!

You’re absolutely right—ProxiFyre connects directly to the IP address instead of the hostname. This behavior occurs because the connection is intercepted and redirected after the hostname has already been resolved—specifically, once the TCP SYN packet has been sent. At that stage, only the destination IP address is available to ProxiFyre, and the original hostname is no longer accessible.
